lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -299,34 +316,39 @@ define((require, exports, module) ->
       )
 
       currentToken = tokenIterator.getCurrentToken()
-      { row: prevRow } = tokenIterator.getCurrentTokenPosition()
+      prevRow = tokenIterator.getCurrentTokenPosition().row
       # TODO: magic string? importing is hard though
       boundaryCorrect = true
+      reason = null
       while LatexParsingContext.isType(currentToken, "equation")
         currentToken = moveToBoundary()
         if not currentToken?
           boundaryCorrect = false
+          reason = "end of a document reached while in math environment"
```

Comment:
since we need i18n, I suggest returning reason identifiers and applying i18n to them in the client code